eng/versioning/set_versions.py

```diff
@@ -249,10 +251,21 @@ def increment_library_version(build_type, artifact_id, group_id):
                         # This is the case where, somehow, the versioning verification has failed and
                         # the prerelease verification doesn't match "beta.X"
                         if prever is None:
-                            raise ValueError('library_to_update ({}:{}) has an invalid prerelease version ({}) which should be of the format beta.X'.format(library_to_update, module.current, vmatch.group('prerelease')))
-                        rev = int(prever.group('revision'))
-                        rev += 1
-                        new_version = '{}.{}.{}-beta.{}'.format(vmatch.group('major'), vmatch.group('minor'), vmatch.group('patch'), str(rev))
+                            # if the build_type isn't data then error
+                            if build_type.name.lower() != 'data':
+                                raise ValueError('library_to_update ({}:{}) has an invalid prerelease version ({}) which should be of the format beta.X'.format(library_to_update, module.current, vmatch.group('prerelease')))
+                            else:
+                                # verify that prerelease is "beta"
+                                if prerelease_data_regex.match(vmatch.group('prerelease')) is None:
+                                    raise ValueError('library_to_update ({}:{}) has an invalid prerelease version ({}) which should be of the format (beta) or (beta.X)'.format(library_to_update, module.current, vmatch.group('prerelease')))
+                                # in the case there the prerelease version is just "beta", increment the minor and set the patch to 0
+                                minor = int(vmatch.group('minor'))
```

Comment:
Do we even know if this is the pattern then want for track 1? I kind of feel like if they don't want to follow the full guidance then we should simply not have support for incrementing their versions. Maybe we should just ignore version increments for those.

Reply:
1.0.0-beta turning into 1.1.0-beta at least follows the same GA increment pattern. For the most part I'm expecting the version to be changed before the next release anyways. What matters for the auto increment is that we produce something that won't clash with what's on Maven and is a higher version. I don't want to ignore the auto-increment for track 1 because a couple of libraries were allowed to be exceptions when most of the others are following the version scheme.